assets/src/state.ts

```diff
@@ -532,9 +561,21 @@ const routeTabsReducer = (
           newRouteTab(state.length),
         ],
       ]
+    case "CREATE_PRESET":
+      return [state, [...state, newPreset(action.payload.sourceTab)]]
```

Comment:
**question:** See below comment, but one thing to consider about this interaction - this means that your existing tab will stay as "untitled" rather than becoming an instantiation of the preset you just created. Are we sure that's what we want?